### PR TITLE
feat: add limit flag

### PR DESCRIPTION
**Pull request by slayer321**
Signed-off-by: slayer321 <sachin.maurya7666@gmail.com>

add limit flag
```
--limit
```

related issue #40 

cc @nyrahul @daemon1024

### CODE REVIEW

**Review by daemon1024 — COMMENTED** [review]
I am a bit skeptical about this approach. Would Atomic Counters be more feasible here and can avoid this much refactoring.

File: log/logClient.go [hunk 1]
```diff
@@ -202,124 +210,143 @@ func regexMatcher(filter *regexp.Regexp, res string) bool {
...
 			}
+			_ = watchAlertsHelper(res, o)
 
-			str = str + fmt.Sprintf("Result: %s\n", res.Result)
 		}
+		Limitchan <- true
+
+	} else {
+		for fd.Running {
+			res, err := fd.alertStream.Recv()
```
Comment (nyrahul) [inline log/logClient.go:343]: Is the possible to remove this code duplication ... I mean `Recv() ... watchAlertsHelper()`

**Review by nyrahul — APPROVED** [review]
LGTM. One comment provided inline. Please check if it can be handled.

**Comment by nyrahul** [comment]
Can you please fix the conflict by rebasing and please check the review comment ? Thanks

File: log/logClient.go (hunk 1 above)
Comment (slayer321) [reply log/logClient.go:343]: I think currently only code duplication section is this one 
```
res, err := fd.logStream.Recv()
if err != nil {
	fmt.Printf("Failed to receive an alert (%s)\n", err.Error())
	break
}
_ = WatchLogsHelper(res, o)
```

I tried to add it in another function ... but it didn't help

**Review by daemon1024 — APPROVED** [review]

**Pull request merged**